libs/clients/middlewares/src/lib/withCache/withCache.ts

```diff
@@ -257,29 +259,37 @@ export function withCache({
   return fetchWithCache
 }
 
-function defaultCacheKey(request: Request) {
+export function defaultCacheKey(request: Request) {
   // Here we trim the origin and the protocol from the URL to clean up cache
   // keys and increase cache hits in environments where the same API is routed
   // in different ways.
   // To avoid cache collisions, we additionally prefixed the cache key with the
   // Enhanced Cache name.
   const url = new URL(request.url)
-  return `${url.pathname}${url.search}`
+
+  // Add automatic calculation of cacheable headers value
+  const headersCacheKey = calculateHeadersCacheKey(request.headers)
+
+  return `${url.pathname}${url.search}${headersCacheKey}`
 }
 
-/**
- * Create a cache key that includes a header value based on a header name if header is present.
- *
- * This is required for clients that are using cache and the URL contains a static placeholder
- * for the resource identifier, e.g. /users/.national-id where the actual resource identifier is
- * passed in a header.
- */
-export function defaultCacheKeyWithHeader(header: string) {
-  return (request: Request) => {
-    const cacheKey = defaultCacheKey(request)
-    const paramHeader = request.headers.get(header)
-    return paramHeader ? `${cacheKey}#${paramHeader}` : cacheKey
+export function calculateHeadersCacheKey(requestHeaders: Headers): string {
+  const cacheableHeadersValues: string[] = []
+  const lowerCasedPatterns = COMMON_HEADER_PATTERNS.map((pattern) =>
+    pattern.toLowerCase(),
+  )
+
+  for (const [name, value] of requestHeaders) {
+    for (const pattern of lowerCasedPatterns) {
+      if (name.toLowerCase().startsWith(pattern)) {
+        cacheableHeadersValues.push(value)
```

Comment:
Should include something like `name.toLowerCase().slice(pattern.length)` in the cache key, as well as escape "#", to avoid collisions and possible attacks.

Reply:
I have added this

```
const nameLower = name.toLowerCase()
if (nameLower.startsWith(pattern)) {
  cacheableHeadersValues.push(
    `${nameLower}=${value.replace(
      CACHE_KEY_SEPARATOR,
      CACHE_KEY_SEPARATOR.repeat(2),
    )}`,
  )
}
```